### DISCUSSION TITLE
Clіckhouse datasourse to Grafana

### DISCUSSION THREAD

**Discussion by Kadadg** [akvorado/akvorado#1331] [Q&A] [answered]
Hi
Question is can i use data in clickhouse to display it in external grafana server?
Need custom display for NAT IPFIX

I not good in docker and log\pass, access to clickhouse not provided in doc

**Comment by coestoes** [comment]
Hello Kadadg,
I am also stuck trying to connect Grafana to Akvorado's Clickhouse database. Docker does not expose the Clickhouse port so you cannot directly connect to Clickhouse. I have tried to install Clickhouse on a seperate VM that Grafana can also access but an external databases is not compatible with Akvorado. Additionally, there is a Grafana directory within Akvorado but I am not sure how it helps. The discussion 1080 (https://github.com/akvorado/akvorado/discussions/1080) briefly goes over the process but I haven't gotten around to try it.

**Reply by vincentbernat** [reply, marked as answer]
You can uncomment the monitoring stack in `.env` to get a Grafana install. This is not really advertised yet as there is no documentation around it and I plan to include a better integration with Grafana in the future. Alternatively, you can override some settings with `docker-compose.override.yml`, notably expose the ClickHouse port for use with an external Grafana.

**Reply by Kadadg** [reply]
Worked for me
in docker-compose.yml added in clickhouse section
     ports:
      - 9000:9000/tcp
